megamek/src/megamek/utils/BoardsTagger.java

```diff
@@ -267,26 +268,36 @@ private static void tagBoard(File boardFile) {
         matchingTags.put(Tags.TAG_SNOWTERRAIN, (snowTerrain > normSide * 2));
         matchingTags.put(Tags.TAG_FLAT, (levelExtent <= 2) && (weighedLevels < normSide * 5));
 
-        // Remove any auto tags that might be present so that auto tags that no longer apply
+        // Remove (see below) any auto tags that might be present so that auto tags that no longer apply
         // are not left in the board file.
-        List<String> toRemove = board.getTags().stream().filter(t -> t.contains(AUTO_SUFFIX)).collect(toList());
-        toRemove.forEach(board::removeTag);
+        Set<String> toRemove = board.getTags().stream().filter(t -> t.contains(AUTO_SUFFIX)).collect(toSet());
 
-        // Give the board any applicable tags
-        matchingTags.keySet().stream().filter(t -> matchingTags.get(t)).map(Tags::getName).forEach(board::addTag);
+        // Find any applicable tags to give the board
+        Set<String> toAdd = matchingTags.keySet().stream().filter(t -> matchingTags.get(t)).map(Tags::getName).collect(toSet());
 
         if (DEBUG) {
             System.out.println("----- Board: " + boardFile);
-            matchingTags.keySet().stream().forEach(System.out::println);
+            if (toRemove.equals(toAdd)) {
+                System.out.println("No changes");
+            } else {
+                toAdd.stream().forEach(System.out::println);
+            }

```

Comment:
MegaMek Logger swapover

Reply:
The tagger is a standalone tool. Seems to make sense to report to the console directly.

Reply:
Ah, if this isn't for the logger these can be ignored (I tend to assume logger)